src/renderer/components/wallet/txs/upgrade/Upgrade.tsx

```diff
@@ -0,0 +1,435 @@
+    successActionHandler,
+    reloadFeeHandler,
+    reloadBalancesHandler
+  } = props
+
+  const intl = useIntl()
+
+  const [form] = Form.useForm<FormValues>()
+
+  // State for visibility of Modal to confirm upgrade
+  const [showConfirmUpgradeModal, setShowConfirmUpgradeModal] = useState(false)
+  // (Possible) subscription of upgrade tx
+  const [upgradeTxSub, _setUpgradeTxSub] = useState<O.Option<Rx.Subscription>>(O.none)
+
+  // unsubscribe upgrade$ subscription
+  const unsubscribeUpgradeTxSub = useCallback(() => {
+    FP.pipe(
+      upgradeTxSub,
+      O.map((sub) => sub.unsubscribe())
+    )
+  }, [upgradeTxSub])
+
+  const setUpgradeTxSub = useCallback(
+    (state) => {
+      unsubscribeUpgradeTxSub()
+      _setUpgradeTxSub(state)
+    },
+    [unsubscribeUpgradeTxSub]
+  )
+
+  useEffect(() => {
+    // Unsubscribe of (possible) previous subscription of `deposit$`
+    return () => {
+      unsubscribeUpgradeTxSub()
+    }
+  }, [unsubscribeUpgradeTxSub])
+
+  // State of upgrade tx
+  const [upgradeTxState, setUpgradeTxState] = useState<UpgradeRuneTxState>(INITIAL_UPGRADE_RUNE_STATE)
+
+  const resetTxState = useCallback(() => {
+    setUpgradeTxState(INITIAL_UPGRADE_RUNE_STATE)
+    setUpgradeTxSub(O.none)
+  }, [setUpgradeTxSub])
+
+  const onFinishHandler = useCallback(() => {
+    reloadBalancesHandler()
+    resetTxState()
+  }, [reloadBalancesHandler, resetTxState])
+
+  const [amountToUpgrade, setAmountToUpgrade] = useState<BaseAmount>(ZERO_BASE_AMOUNT)
+
+  const oRuneBnbAmount: O.Option<BaseAmount> = useMemo(
+    () => FP.pipe(oBalances, O.chain(FP.flow(getRuneBnBAmountFromBalances, O.map(assetToBase)))),
+    [oBalances]
+  )
+
+  const oBnbAmount: O.Option<BaseAmount> = useMemo(
+    () => FP.pipe(oBalances, O.chain(FP.flow(getBnbAmountFromBalances, O.map(assetToBase)))),
+    [oBalances]
+  )
+
+  const maxAmount: BaseAmount = useMemo(
+    () =>
+      FP.pipe(
+        oRuneBnbAmount,
+        O.getOrElse(() => ZERO_BASE_AMOUNT)
+      ),
+    [oRuneBnbAmount]
+  )
+
+  useEffect(() => {
+    setAmountToUpgrade(maxAmount)
+  }, [maxAmount])
+
+  useEffect(() => {
+    // Whenever `amountToUpgrade` has been updated, we put it back into input field
+    form.setFieldsValue({
+      amount: baseToAsset(amountToUpgrade).amount()
+    })
+  }, [amountToUpgrade, form])
+
+  const amountValidator = useCallback(
+    async (_: unknown, value: BigNumber) => {
+      console.log('validator:', value.toString())
+      return validateTxAmountInput({
+        input: value,
+        maxAmount: baseToAsset(maxAmount),
+        errors: {
+          msg1: intl.formatMessage({ id: 'wallet.errors.amount.shouldBeNumber' }),
+          msg2: intl.formatMessage({ id: 'wallet.errors.amount.shouldBeGreaterThan' }, { amount: '0' }),
+          msg3: intl.formatMessage({ id: 'wallet.errors.amount.shouldBeLessThanBalance' })
+        }
+      })
+    },
+    [intl, maxAmount]
+  )
+
+  const onChangeInput = useCallback(
```

Comment:
pls merge latest master. we can not input decimal values right now. i remember there was a fix for that

Reply:
Ok